miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/SampleService.java

```diff
@@ -34,9 +34,11 @@ default EntityType getEntityType() {
 
   Collection<SampleIdentity> getIdentitiesByExternalNameOrAlias(String externalName) throws IOException;
 
-  void confirmExternalNameUniqueForProjectIfRequired(String externalNames, Sample sample) throws IOException, ConstraintViolationException;
+  Collection<SampleIdentity> getIdentitiesByExactExternalName(String externalName) throws IOException;
+
+  Collection<SampleIdentity> getIdentitiesByExactExternalNameAndProject(String externalName, Long projectId) throws IOException;
 
-  Collection<SampleIdentity> getIdentitiesByExternalNameAndProject(String externalName, Long projectId) throws IOException;
+  void confirmExternalNameUniqueForProjectIfRequired(String externalNames, Sample sample) throws IOException, ConstraintViolationException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.hibernate.exception.ConstraintViolationException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)